File: Wikipedia/Code/UIView+WMFWelcomeFadeInAndUp.m
#import "UIView+WMFWelcomeFadeInAndUp.h"

@implementation UIView (WMFWelcomeFadeInAndUp)

- (void)wmf_zeroLayerOpacity {
    self.layer.opacity = 0.0;
}

- (void)wmf_fadeInAndUpAfterDelay:(CGFloat)delay {
    self.layer.transform = CATransform3DMakeTranslation(0, 18, 0);
    [UIView animateWithDuration:0.4
                          delay:delay
                        options:UIViewAnimationOptionCurveEaseInOut
                     animations:^{
                       self.layer.opacity = 1.0;
                       self.layer.transform = CATransform3DIdentity;
                     }
                     completion:nil];
}

@end

File: Wikipedia/Code/UIView+WMFWelcomeFadeInAndUp.swift

extension UIView {
    func wmf_zeroLayerOpacity() {
        layer.opacity = 0
    }

    func wmf_fadeInAndUpWithDuration (_ duration: Double, delay: Double) {
        layer.transform = CATransform3DMakeTranslation(0, 18, 0)
        UIView.animate(
            withDuration: duration,
            delay: delay,
            options: UIView.AnimationOptions(),
            animations: {
                self.layer.opacity = 1
                self.layer.transform = CATransform3DIdentity
            }, completion: nil)
    }
}
